utils/eigenda/host/src/witness_generator.rs

```diff
@@ -0,0 +1,157 @@
+    ) -> Result<Self::WitnessData> {
+        let preimage_witness_store = Arc::new(std::sync::Mutex::new(PreimageStore::default()));
+        let blob_data = Arc::new(std::sync::Mutex::new(BlobData::default()));
+
+        let preimage_oracle = Arc::new(kona_proof::CachingOracle::new(
+            2048,
+            OracleReader::new(preimage_chan),
+            HintWriter::new(hint_chan),
+        ));
+        let blob_provider = OracleBlobProvider::new(preimage_oracle.clone());
+
+        let oracle = Arc::new(PreimageWitnessCollector {
+            preimage_oracle: preimage_oracle.clone(),
+            preimage_witness_store: preimage_witness_store.clone(),
+        });
+        let beacon = OnlineBlobStore { provider: blob_provider.clone(), store: blob_data.clone() };
+
+        // Create EigenDA blob provider that collects witness data
+        let eigenda_blob_provider = OracleEigenDAProvider::new(oracle.clone());
+        let eigenda_blobs_witness = Arc::new(Mutex::new(EigenDABlobWitnessData::default()));
+
+        let eigenda_blob_and_witness_provider = OracleEigenDAWitnessProvider {
+            provider: eigenda_blob_provider,
+            witness: eigenda_blobs_witness.clone(),
+        };
+
+        let executor = EigenDAWitnessExecutor::new(eigenda_blob_and_witness_provider);
+
+        let (boot_info, input) = get_inputs_for_pipeline(oracle.clone()).await.unwrap();
+        if let Some((cursor, l1_provider, l2_provider)) = input {
+            let rollup_config = Arc::new(boot_info.rollup_config.clone());
+            let pipeline = WitnessExecutorTrait::create_pipeline(
+                &executor,
+                rollup_config,
+                cursor.clone(),
+                oracle.clone(),
+                beacon,
+                l1_provider.clone(),
+                l2_provider.clone(),
+            )
+            .await
+            .unwrap();
+            WitnessExecutorTrait::run(&executor, boot_info.clone(), pipeline, cursor, l2_provider)
+                .await
+                .unwrap();
+        }
+
+        // Extract the EigenDA witness data
+        let mut eigenda_witness_data = std::mem::take(&mut *eigenda_blobs_witness.lock().unwrap());
+
+        // Generate canoe proofs using the reduced proof provider for proof aggregation
+        use canoe_sp1_cc_host::CanoeSp1CCReducedProofProvider;
+        let canoe_provider = CanoeSp1CCReducedProofProvider {
+            eth_rpc_url: std::env::var("L1_RPC").ok().unwrap_or_default(),
+        };
+        let canoe_proofs = hokulea_witgen::from_boot_info_to_canoe_proof(
+            &boot_info,
+            &eigenda_witness_data,
+            oracle.clone(),
+            canoe_provider,
+        )
+        .await?;
+
+        // Store the canoe proof in the witness data
+        let canoe_proof_bytes =
+            serde_cbor::to_vec(&canoe_proofs).expect("Failed to serialize canoe proof");
+        eigenda_witness_data.canoe_proof_bytes = Some(canoe_proof_bytes);
```

Comment:
Seems this shouldn't go into the witness that is read into the VM

https://github.com/Layr-Labs/hokulea/blob/master/crates/proof/src/canoe_verifier/sp1_cc.rs#L53

Reply:
Made a fix [here](https://github.com/succinctlabs/op-succinct/pull/572/commits/ec638a47c8d6fa44f47b1202c2a79e222628c0db)